mixer/pkg/runtime/dispatcher/dispatcher.go

```diff
@@ -116,21 +117,23 @@ const (
 )
 
 // Check implementation of runtime.Impl.
-func (d *Impl) Check(ctx context.Context, bag attribute.Bag) (adapter.CheckResult, error) {
+func (d *Impl) Check(ctx context.Context, bag attribute.Bag) (adapter.CheckResult, *v1.RouteDirective, error) {
 	s := d.getSession(ctx, tpb.TEMPLATE_VARIETY_CHECK, bag)
 
 	var r adapter.CheckResult
+	var dir *v1.RouteDirective
 	err := s.dispatch()
 	if err == nil {
 		r = s.checkResult
+		dir = s.directive
 		err = s.err
 
 		if err == nil {
 			// No adapters chimed in on this request, so we return a "good to go" value which can be cached
 			// for up to a minute.
 			//
 			// TODO: make these fallback values configurable
-			if r.IsDefault() {
+			if adapter.IsDefault(&r) {
```

Comment:
I suggest changing the parameter type to pass-by-value in order to avoid creating temp objects in the heap.

Reply:
In general, this runtime dispatching code is being very careful about memory allocations for the sake of efficiency. Does returning *RouteDirective result in memory allocations? Could we eliminate that?